public/app/plugins/datasource/cloud-monitoring/datasource.ts

```diff
@@ -163,11 +163,13 @@ export default class CloudMonitoringDatasource extends DataSourceWithBackend<
           });
         }),
         map(({ data }) => {
-          return data;
-        }),
-        map((response) => {
-          const result = response.results[refId];
-          return result && result.meta ? result.meta.labels : {};
+          const dataQueryResponse = toDataQueryResponse({
+            data: data,
+          });
+          if (dataQueryResponse.data.length !== 0 && dataQueryResponse.data[0].meta) {
```

Comment:
This you should be able to replace this with:
`return dataQueryResponse?.data[0]?.meta?.custom?.labels ?? {}`